content/en/blog/_posts/2022-08-26-Announcing-Auto-Refreshing-Official-CVE-Feed/index.md

```diff
@@ -0,0 +1,159 @@
+---
+layout: blog 
+title: "Announcing the Auto-refreshing Official Kubernetes CVE
+Feed"
+date: 2022-08-26 
+slug: k8s-cve-feed-alpha
+---
+
+**Author**: Pushkar Joglekar (VMware)
+
+A programmatic way to keep track of Kubernetes CVEs for End Users, has been a
+longstanding request for the Kubernetes Community. With the release of v1.25,
+we are excited to announce availability of such
+a [feed](docs/reference/issues-security/official-cve-feed/) as
+an `alpha` feature. This blog will cover the background, scope and
+details on how this feature was implemented.
+
+## Motivation
+
+With the growing number of eyes on Kubernetes, the number of CVEs related to
+Kubernetes have increased. Although most CVEs that directly, indirectly, or
+transitively impact Kubernetes are regularly fixed, there is no single place for
+the end users of Kubernetes to programmatically subscribe or pull the data of
+fixed CVEs. Current options are either broken or incomplete.
+
+## Scope
+
+### Goals
+
+Create a periodically auto-refreshing, human and machine-readable list of
+official Kubernetes CVEs
+
+### Non-Goals
+
+* Triage and vulnerability disclosure will continue to be done by SRC (Security Response
+Committee). 
+* Listing CVEs that are identified in build time dependencies and container
+images are out of scope. 
+* Only official CVEs announced by the Kubernetes SRC will be published in the feed.
+
+### Personas and User stories
+
+* **End Users**: Persons or teams who _use_ Kubernetes to deploy applications
+  they own
+* **Platform Providers**: Persons or teams who _manage_ Kubernetes clusters
+* **Maintainers**: Persons or teams who _create_ and _support_ Kubernetes
+  releases through their work in Kubernetes Community - via various Special 
+  Interest Groups and Committees.
+
+## Implementation Details
+
+An auto-refreshing CVE feed allows End Users to programmatically fetch the list
+of CVEs and allows them to get the latest information from Kubernetes Community.
+
+The feed is updated to ensure freshness and minimal maintainer overhead
+by fetching the CVE related information through existing SRC workflow of
+creating a GitHub Issue when announcing new Kubernetes CVEs.
+
+### Pre-requisites
+
+Until December 2021, it was not possible to filter for issues or PRs that are
+tied to CVEs announced by Kubernetes SRC. To address that a new
+label `official-cve-feed` was added. Relevant issues were then labelled with this
+new label. The in-scope issues are `closed` issues for which
+there is a CVE ID(s) and is officially announced as a Kubernetes security
+vulnerability by SRC. You can now filter on all of these issues and
+find them 
+[here](https://github.com/kubernetes/kubernetes/issues?q=is%3Aissue+is%3Aclosed+label%3Aofficial-cve-feed+).
+
+For future security vulnerabilities, this label was added to the SRC playbook
+such that all the future in-scope issues will automatically have this label.
+
+### Prow, Cloud Buckets and Netlify - ftw
```

Comment:
```suggestion
### Prow, Cloud Buckets and Netlify {#implementation}
```

Reply:
> You know what, I'd split this into two articles:
> 
> * https://blog.k8s.io/ article: **Announcing the Official Kubernetes CVE Feed**
> * https://k8s.dev/blog/ article: **How I Implemented the Kubernetes Security Issues Feed**
> 
> What do you think @PushkarJ ? If we do that, both articles can go out on the same day and link to each other.

That's an interesting idea. Which sections in the current file would you recommend moving to the second blog on k8s.dev ?

Reply:
> Which sections in the current file would you recommend moving to the second blog on k8s.dev ?

I'd shift all of the following to the contributor blog article:

```markdown
## Implementation Details

…

### Pre-requisites

…

### Prow, Cloud Buckets and Netlify 

…

## Design Considerations

…

### Integrity and Access Control Protections

…

#### GitHub Issues for Publicly announced CVEs

…

#### Prow configuration

…

#### Google Cloud Bucket

…

#### Website data templates
```

I would put the same
```markdown
### Freshness guarantees
```
heading in **both** articles, but with different content: outcome vs. implementation.

Reply:
@sftim makes sense to me. This would require some additional work though, so want to be cognizant of any deadlines. Are we okay if we post the feature blogs few days after v1.25 is GA (Aug 26) ? I don't expect to be able to work on this refactor until Monday Aug 22

Reply:
I recommend that you focus on the announcement article - keep it simple - and work on the contributor article if and as time allows.
You already planned to have the announcement go out _after_ the new feed goes live, right?